### ISSUE TITLE
[Suggestion] Consider using react-native-web

### ISSUE THREAD

**Issue by immortalx** [Urigo/WhatsApp-Clone-Client-React#51]
I believe its a good fit for the project and react-native-web would benefit from the production ready example.

**Comment by Urigo** [comment]
I love it!
The idea for those tutorials is that we can compare different tools and stacks on a full scale app.

Maybe you can help start a version of this, link it here and we can start iterating on it?

Also, I want to add tools that measure things so we could provide more clear picture comparing tools.

What would you say would be things we should highlight when comparing the current implementation with react-native-web?
Then let's see if we can also add them to the CI to track them over time!
From the top of my head:
* Compare bundle size
* Lighthouse stuff
* Number of lines of code - I don't think number of lines of code is a good measurement, but we will provide a code diff between versions of the tutorial and give people a clear picture of the differences in the implementation
* Add a chapter about creating a native mobile app and compare the effort

To be honest, I never tried react-native-web myself, so it might be a good place to start.

Some questions I have (please don't kill me if those are really dumb questions):
* I see that `react-native-web` is currently compatible with React Native 0.55 where the latest stable version is 0.58. As I understand, Hooks will probably land in React Native not earlier then 0.60. I don't mind having a version of the tutorial without hooks, that can be one of the tradeoffs people would choose from. 
* How does it compare to [react-native-dom](https://github.com/vincentriemer/react-native-dom)? should we talk about it here or create a separate issue for it? are there things we could track to compare those as well?

By the way, if anyone wants to participate in that effort, you can write here or email me directly

**Comment by alanpog** [comment]
@Urigo, RN can be made to work with hooks now, see this issue: https://github.com/facebook/react-native/issues/21967

**Comment by immortalx** [comment]
The main features and benefits are well defined by the author on the project page, personally i really find it easier to work within react native design constraints. The library was designed for the twitter lite project https://blog.twitter.com/engineering/en_us/topics/open-source/2017/how-we-built-twitter-lite.html , http://mobile.twitter.com and is in the process to become the main twitter site. It powers the PWA android app as well.

The version twitter uses is heavily customized and has some solutions that are not in the open sourced version, i don't know if they plan to PR those optimizations and new features in the future.

- Number of lines of code should be the same or a little bit less lines.
- Since we are recreating WhatsApp UI, it's a good thing that we can share components between native and web.
- The only pain point that i can remember is dealing with nested scrolls or just scrolling in general, but for WhatsApp UI i don't think it's an issue. 
- react-native-dom is still experimental but I'm very excited about it, i would consider it when it reaches beta. I believe it will be faster and even easier to share components because of the way it's designed.

**Comment by immortalx** [comment]
@alanpog i don't think we can use hooks with react-native-web yet, it's still on 0.55 and next release will be compatible with 0.57. I can see a lot of people requesting hooks once it lands so support should be coming soon.

**Comment by ankibalyan** [comment]
`react-native@0.59` will be compatible with `hooks`. 😄

**Comment by alanpog** [comment]
A RC of RN with hooks was released today: https://github.com/facebook/react-native/releases/tag/v0.59.0-rc.1
We still need `react-native-web` to support it though.

**Comment by Urigo** [comment]
What do you think about other libraries like https://react-native-training.github.io/react-native-elements/ ?
maybe maintainers from those libraries would like to take on the challenge of show it's possible to create a real cross platform app using their tools?

**Comment by veerjainATgmail** [comment]
would react-native-web version be easy to convert it into a react-native mobile app ?

**Comment by Urigo** [comment]
@veerjainATgmail yes that is the point of the transition.

I would really love someone to try to clone the master and try those things out and give us feedback on that.
That is the next most important feature I want to focus on

**Comment by cybercoder** [comment]
Main challenge is creating global subscription to update all active conversations, not only which conversation we are in at time.  ( don't use subscribeTomore in the room ).
updating cache  and creating cached item if there is not is another problem. I'm working on it but i have no idea how i can handle and control app when using persist-cache to keep data offline like telegram messaging and others. Another challenge is writing chek-mark and double check-mark on sent messages and seen messages.

**Comment by cybercoder** [comment]
I found a solution can be turning Apollo cache entirely off and using ```realm``` as offline database to save conversations & messages in phone store, update it on subscriptions and in conversation screen mount time (useMemo) fetch messages of conversation which they are greater than last seen message of that conversation by user and save it in ```realm``` store. Also we can find differences in ```realm```and server database.
It seems ```react-navigation```is not a goof choice (may i am wrong) for creating navigation because control of upper pages is difficult. ( A drawer will re-render every thing on it when you swipe it!). So i found ```react-native-navigation``` is a better way.
